src/parser/csql_grammar.y

```diff
@@ -21587,45 +21557,129 @@ opt_encrypt_algorithm
       $$ = 2; }   /* TDE_ALGORITHM_ARIA */
   ;
 
-opt_deduplicate_key_mode
-        :  /* empty */
-	       { DBG_TRACE_GRAMMAR(opt_deduplicate_key_mode, : ); 
-                 $$ = COMPRESS_MODE_NOT_SET;
-               }
-        | DEDUPLICATE_KEY OFF_
-               { DBG_TRACE_GRAMMAR(opt_deduplicate_key_mode,  | DEDUPLICATE_KEY OFF_); 
-                 $$ = MAKE_COMPRESS_MODE_LEVEL(DEDUPLICATE_KEY_MODE_OFF, DEDUPLICATE_KEY_LEVEL_NONE);
-               }
-        | DEDUPLICATE_KEY ON_ opt_deduplicate_key_level
-               { DBG_TRACE_GRAMMAR(opt_deduplicate_key_mode,  | DEDUPLICATE_KEY MEDIUM_ opt_deduplicate_key_level); 
-                 $$ = $3;
-               }        
+opt_fk_deduplicate_off
+        : /* empty */
+          { DBG_TRACE_GRAMMAR(opt_fk_deduplicate_off, : );
+            $$ = false; }
+        | WITH DEDUPLICATE_ '=' unsigned_integer
+          { DBG_TRACE_GRAMMAR(opt_fk_deduplicate_off,  | DEDUPLICATE_ '=' unsigned_integer); 
+           /* It can be entered through the table constraint clause of "CREATE TABLE" and the add constraint clause of "ALTER TABLE".
+            * If it is entered in the "CREATE TABLE" statement, an error is processed.
+           */          
+            if(parser_is_null_alter_node())
```

Comment:
Is the purpose of preventing developers from adding this rule elsewhere?

Reply:
For FK, only "with deduplicate=0" can be specified using the ALTER TABLE statement. However, it is not a feature that is open to the general user.For FK, only "with deduplicate=0" can be specified using the ALTER TABLE statement. However, it is not a feature that is open to the general user.

Reply:
I was mistaken that this rule is not used for create_stmt. Thank you